main.go

```diff
@@ -10,22 +10,28 @@ import (
 	"github.com/okkur/reposeed/cmd/reposeed/config"
 )
 
+const SupportedConfigVersion = "v1"
+
 func main() {
 	err := godotenv.Load()
 	if err != nil {
 		log.Println("Couldn't find .env file. Reading environment variables from system")
 	}
 	app := gin.Default()
 	config := &config.Config{}
-	app.POST("/generate", func(ctx *gin.Context) {
-		ctx.BindJSON(config)
-		filename, err := generator.CreateFiles(*config, config.Project.Name, os.Getenv("STORAGE"))
-		if err.Code != 200 {
-			ctx.JSON(400, err)
-			ctx.Abort()
-		} else {
-			ctx.File(filename)
-		}
-	})
-	app.Run(os.Getenv("PORT"))
+	if config.Project.Version == SupportedConfigVersion {
```

Comment:
app.POST() defines an endpoint for the API. It shouldn't be inside an if statement.

Reply:
Also in this step, ``config`` is empty, we fill the ``config`` with information that we receive in an HTTP request. (Line 24).